### PR TITLE
feat: add library for handling request validation

### LINKED ISSUE #334: Improve Request Validation
Use a third party package to handle request validation. Wrap all validation errors together and return to client at once.

### PR DESCRIPTION
**Pull request by Dotunj**
Resolves #334 #290

### CODE REVIEW

File: util/validator.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+func init() {
+	govalidator.TagMap["supported_hash"] = govalidator.Validator(func(hash string) bool {
+		if _, ok := algo.M[hash]; !ok {
+			return false
+		}
+
+		return true
+	})
```
Comment (danvixent) [inline util/validator.go:38]: I think this can be somewhere in main.go

Comment (Dotunj) [reply util/validator.go:38]: I thought about doing that too, but my reason for doing this is just incase we need to add more custom validation down the line, it's cleaner and clearer this way. Any downsides to doing it this way? @danvixent

File: config/config.go [hunk 2]
```diff
@@ -67,18 +67,18 @@ type AuthConfiguration struct {
 }
 
 type StrategyConfiguration struct {
-	Type    StrategyProvider             `json:"type"`
-	Default DefaultStrategyConfiguration `json:"default"`
+	Type    StrategyProvider             `json:"type" valid:"required~please provide a valid strategy type,in(default)~unsupported strategy type"`
```
Comment (Subomi) [inline config/config.go:70]: @Dotunj This looks like we're coupling configuration validation with server validation. I don't think we want to do this.

File: server/group_test.go [hunk 3]
```diff
@@ -121,7 +121,22 @@ func TestApplicationHandler_CreateGroup(t *testing.T) {
 	ctrl := gomock.NewController(t)
 	defer ctrl.Finish()
 
-	bodyReader := strings.NewReader(`{"name": "ABC_DEF_TEST_UPDATE"}`)
+	bodyReader := strings.NewReader(`{
```
Comment (Subomi) [inline server/group_test.go:124]: @Dotunj There should be more tests for this guy. Like catch validation errors using a test table.

File: util/validator.go (hunk 1 above)
Comment (jirevwe) [reply util/validator.go:38]: I prefer it here, let's not litter main with so much logic.

**Review by jirevwe — APPROVED** [review]

File: util/validator.go (hunk 1 above)
Comment (Subomi) [reply util/validator.go:38]: This is probably fine.

**Review by Subomi — CHANGES_REQUESTED** [review]
FYI @Dotunj

File: config/config.go (hunk 2 above)
Comment (Dotunj) [reply config/config.go:70]: @Subomi Are you referring to just this particular field?

**Pull request closed without merging**